nano/store/iterator.cpp

```diff
@@ -1 +1,104 @@
+#include <nano/lib/utility.hpp>
 #include <nano/store/iterator.hpp>
+
+namespace nano::store
+{
+void iterator::update ()
+{
+	std::visit ([&] (auto && arg) {
+		using T = std::remove_cvref_t<decltype (arg)>;
+		if constexpr (std::is_same_v<T, lmdb::iterator>)
+		{
+			if (!arg.is_end ())
+			{
+				auto & current = *arg;
+				std::span<uint8_t const> key{ reinterpret_cast<uint8_t const *> (current.first.mv_data), current.first.mv_size };
+				std::span<uint8_t const> value{ reinterpret_cast<uint8_t const *> (current.second.mv_data), current.second.mv_size };
+				this->current = std::make_pair (key, value);
```

Comment:
It seems wrong to include this code here, similar to all the other methods this should be a simple:
```
std::visit ([] (auto && arg) {
 	this->current = arg.current ();
 },
 ```
 With the database specific code hidden in a particular backend implementation.

Reply:
I added a span() function on the concrete iterators. Internally these want to hold their native pointers for iteration e.g. MDB_val and rocksdb::Slice but this way the database-specific part is in the database-specific iterator.